### DISCUSSION TITLE
Can't Unresolved reference Bigdecimal. But I added ExtendedScalars

### DISCUSSION THREAD

**Discussion by GodChiken** [Netflix/dgs-framework#601] [Q&A] [answered]
Hi
i have some problem when i use netflix dgs.

I added some type ExtendsScalar in api-module

    @DgsComponent
    class ScalarExtendRegistration {
        @DgsRuntimeWiring
        fun addScalar(builder: RuntimeWiring.Builder): RuntimeWiring.Builder {
            return builder.scalar(ExtendedScalars.Date)
                          .scalar(ExtendedScalars.DateTime)
                          .scalar(ExtendedScalars.GraphQLLong)
                          .scalar(ExtendedScalars.GraphQLBigDecimal)
        }
    }

Date, DateTime, GraphpQLLong.. It's well-generated.

But GraphQLBigDecimal.. I checked in this generated code - Unresolved reference: BigDecimal. Because it is not automatically imported.

```public data class Test(
  @JsonProperty("id")
  public val id: Long? = null,
  @JsonProperty("insertedAt")
  public val insertedAt: LocalDate? = null,
  @JsonProperty("updatedAt")
  public val updatedAt: LocalDate? = null,
  @JsonProperty("secret")
  public val secret: String? = null,
  @JsonProperty("name")
  public val name: String? = null,
  @JsonProperty("nickName")
  public val nickName: String? = null,
  @JsonProperty("address")
  public val address: String? = null,
  @JsonProperty("money")
  public val money: BigDecimal? = null
) {
  public companion object
}
```

here is my netflixe dgs dependency version
```
Gradle: com.netflix.graphql.dgs.codegen:graphql-dgs-codegen-client-core:5.0.6
Gradle: com.netflix.graphql.dgs:graphql-dgs:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-client:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-extended-scalars:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-mocking:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-spring-boot-oss-autoconfigure:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-spring-webmvc:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-dgs-spring-webmvc-autoconfigure:4.7.3
Gradle: com.netflix.graphql.dgs:graphql-error-types:4.7.3

```

I wonder why BigDecimal is not automatically imported.

**Comment by GodChiken** [comment, marked as answer]
![스크린샷 2021-09-06 오전 10 13 18](https://user-images.githubusercontent.com/16012504/132147514-408d7d97-1b60-4ecd-878c-d5fa6170f5f5.png)

I've added **[ExtendedScalars.GraphQLBigDecimal]** within the Netflix project **[dgs-examples-kotlin]**
, but it's the same thing that happens to my project.

please help me

**Reply by berngp** [reply]
Do you have the mapping in the codgen Gradle task?

**Reply by GodChiken** [reply]
No, you mean this?

![스크린샷 2021-09-06 오후 2 00 59](https://user-images.githubusercontent.com/16012504/132163048-1b715ad3-2c5d-4e0f-9bb7-6edfc4b005ba.png)

**Reply by berngp** [reply]
Yes, you need to define a mapping for BigDecimal.

**Reply by GodChiken** [reply]
Ok. i Fixed

generateJava is cannot be invoked function..
![스크린샷 2021-09-06 오후 2 37 29](https://user-images.githubusercontent.com/16012504/132165846-59c42b33-145d-4374-9210-36794f7d762f.png)

I'm not skilled in kotlin, gradle and netflix dgs.

anyway this is result

```
plugins {
    id("com.netflix.dgs.codegen") version "5.0.6"
}

dependencies {
    implementation(project(":common"))
    implementation(platform("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:4.7.3"))
    implementation("com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter")
    implementation("com.netflix.graphql.dgs:graphql-dgs-extended-scalars")
    implementation("com.fasterxml.jackson.datatype:jackson-datatype-joda")
}

tasks.withType<com.netflix.graphql.dgs.codegen.gradle.GenerateJavaTask> {
    packageName = "com.masterkbh.kotlin.generated"
    typeMapping = mutableMapOf(
        "BigDecimal" to "java.math.BigDecimal",
        "Long" to "java.lang.Long"
    )
}
```

It's working fine. Thanks, bergp.

**Reply by manoj-r** [reply]
@berngp I have a related question. I have been looking at the kotlin example project https://github.com/Netflix/dgs-examples-kotlin.
This project has an extended scalar DateTime, but I don't find the codegen mapping anywhere. How does this work now? And if I try to add a new extended scalar(PositiveInt), the build breaks. Can I get a little more clarity on how extended scalar is working without a mapping?

**Reply by paulbakker** [reply]
@manoj-r For a few types, including DateTime, we have a mapping out of the box in codegen. It's currently a bit arbitrary  which extra types are known by Codegen, which isn't great, but we haven't found a better mechanism yet.
